src/components/hint/index.tsx

```diff
@@ -418,12 +472,11 @@ class Hint extends BaseComponent {
       <React.Fragment>
         {onBackgroundPress ? (
           <Modal
-            pointerEvents="box-none"
             visible={this.showHint}
             animationType="none"
             transparent
             onBackgroundPress={onBackgroundPress}
-            onRequestClose={onBackgroundPress}
+            onRequestClose={onBackgroundPress as () => void}
```

Comment:
Why the change of type to `() => void`?

Reply:
because `onRequestClose` expects a function with `() => void` signature, however `onBackgroundPress` has `(event: GestureResponderEvent) => void`. Therefore I'm adding a type assertion here.

Reply:
This looks like a bug, because the user that sends the `onBackgroundPress` can expect to receive an `event` and may rely on it. I think you need to change the prop to:
```
onBackgroundPress?: (event?: GestureResponderEvent) => void;
```
(and remove the ` as () => void`)

Reply:
I think it's actually the other way around.

`ModalProps` ( in `src/components/modal/index.tsx`) is defined:
```
/**
  * allow dismissing a modal when clicking on its background
  */
onBackgroundPress?: (event: GestureResponderEvent) => void;
```
Whereas `onRequestClose` is defined:
```
/**
  * The `onRequestClose` prop allows passing a function that will be called once the modal has been dismissed.
  * _On the Android platform, this is a required function._
  */
onRequestClose?: () => void;
```

IMO it's important to keep the `onBackgroundPress` definition as is because if the user uses `onBackgroundPress` prop then `event` should always be defined (as you've described). `onRequestClose`, however does not expect a function argument, so we can override it. I think the only reason we use `onRequestClose` is because it's a required function for android (`onBackgroundPress` is what should actually dismiss the modal) . Alternatively, maybe we could even have something like `onRequestClose={this.noop}` (`noop = () => {}`)?

Reply:
Maybe we should take this offline.